### PR TITLE
media-video/mpv: version bump. Wrt bugs #564656 #562394

### PR DESCRIPTION
**Pull request by Vindex17**
https://github.com/mpv-player/mpv/releases/tag/v0.12.0

### CODE REVIEW

File: media-video/mpv/mpv-9999.ebuild [hunk 1]
```diff
@@ -150,8 +148,7 @@ pkg_setup() {
 }
 
 src_prepare() {
-	cp "${DISTDIR}"/waf-${WAF_V} "${S}"/waf || die
-	chmod 0755 "${S}"/waf || die
+	python bootstrap.py || die
```
Comment (maksbotan) [inline media-video/mpv/mpv-9999.ebuild]: This is a bad idea -- bootstrap.py script downloads WAF from internet. I don't think we allow this in ebuilds. Moreover, it won't work with ``FEATURES="network-sandbox"``.

**Comment by maksbotan** [comment]
I've contacted the current proxy maintainer of the package, I hope we'll get new version into the tree soon.
As far as I can tell, this PR introduces some unnecessary style fixes.

File: media-video/mpv/mpv-0.12.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		x11-proto/videoproto
+		xinerama? ( x11-proto/xineramaproto )
+		xscreensaver? ( x11-proto/scrnsaverproto )
+	)
+"
+RDEPEND="${COMMON_DEPEND}
+	selinux? ( sec-policy/selinux-mplayer )
+"
+
+pkg_setup() {
```
Comment (hasufell) [inline media-video/mpv/mpv-0.12.0.ebuild]: this info are all misplaced, they belong in `pkg_pretend` which is the only phase which is carried out _before_ the merge. Anything else can be exptected to go unnoticed and the user will not be able to make an informed choice.

Comment (Vindex17) [inline media-video/mpv/mpv-0.12.0.ebuild]: I just left them as they were in previous ebuilds. Is it ok if I move them to pkg_postinst()?

Comment (hasufell) [inline media-video/mpv/mpv-0.12.0.ebuild]: no, pkg_postinst is not carried out before the merge. We want to give this information to the user as soon as possible.

Comment (Vindex17) [inline media-video/mpv/mpv-0.12.0.ebuild]: Done

File: media-video/mpv/mpv-0.12.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+	libguess? ( >=app-i18n/libguess-1.0 )
+	lua? (
+		!luajit? ( || ( =dev-lang/lua-5.1*:= =dev-lang/lua-5.2*:= ) )
+		luajit? ( dev-lang/luajit:2 )
+	)
+	openal? ( >=media-libs/openal-1.13 )
+	pulseaudio? ( media-sound/pulseaudio )
+	rubberband? ( >=media-libs/rubberband-1.8.0 )
+	samba? ( net-fs/samba )
+	sdl? ( media-libs/libsdl2[threads] )
```
Comment (hasufell) [inline media-video/mpv/mpv-0.12.0.ebuild]: This is missing USE flags. If libsdl2 does not have X, video or sound USE flags enabled, then the mpv sdl support is basically broken.

File: media-video/mpv/mpv-0.12.0.ebuild (hunk 2 above)
Comment (aballier) [inline media-video/mpv/mpv-0.12.0.ebuild]: for 'as soon as possible' we have local use descriptions in metadata.xml, those messages do not add any value IMHO

File: media-video/mpv/mpv-0.12.0.ebuild (hunk 3 above)
Comment (Vindex17) [inline media-video/mpv/mpv-0.12.0.ebuild]: Fixed

File: media-video/mpv/mpv-0.12.0.ebuild (hunk 2 above)
Comment (hasufell) [inline media-video/mpv/mpv-0.12.0.ebuild]: Either they give value (in which case the user wants to see that info _before_ he compiles) or they don't give value and can be removed altogether.

At any rate, pkg_setup and pkg_postinst are both wrong for this kind of information. pkg_preinst is fine.

Comment (Vindex17) [inline media-video/mpv/mpv-0.12.0.ebuild]: I think you are both right. Can I remove them?

Comment (hasufell) [inline media-video/mpv/mpv-0.12.0.ebuild]: That's an intrusive change and should be up to the maintainer. So I'd say no.

**Comment by Vindex17** [comment]
Commit updated for https://github.com/mpv-player/mpv/releases/tag/v0.13.0

**Comment by PPed72** [comment]
It also needs >=x11-libs/libva-1.4.0, otherwise it chokes in configure phase.

**Comment by mgorny** [comment]
@maksbotan, will you handle this?

**Comment by idella** [comment]
Coacher already added as a co-maintainer and he has started looking at the package's open bugs. The current proxy maintainer has shown no activity for a long time

**Comment by Coacher** [comment]
In turn I've contacted @maksbotan via email. Depending on his response delay this PR can take a bit more time before merge unless a Gentoo dev will step up and merge it.

**Comment by Vindex17** [comment]
Commit updated for https://github.com/mpv-player/mpv/releases/tag/v0.14.0

**Comment by papjul** [comment]
mpv >= 0.12.0 requires ffmpeg >= 2.8.3 because of this commit https://github.com/FFmpeg/FFmpeg/commit/4ab56667594842283dc5ae07f0daba2a2cb4d3af which was not included before version 2.8.3.

See note releases for 0.12.0 and 0.13.0 here: https://github.com/mpv-player/mpv/releases

**Comment by Vindex17** [comment]
Updated, hoping someone will accept this pull request, since my efforts here keep being ignored

**Comment by Coacher** [comment]
@Vindex17, I understand your frustration. Sadly, I didn't get any reply from @maksbotan. I guess he is busy with smth else. Another Gentoo dev that maintains mpv, @yngwin, is on devaway atm.

Tomorrow I'll try to contact @idella, who is the leader of the Gentoo proxy-maint team. Hopefully we could work something out.

But the work you've done so far won't go in vain. Thank you.

**Comment by Coacher** [comment]
Ok, I've crafted a 'proper' ebuild for 0.14.0 and updated ebuild for 0.11.0. They include more fixes than the one proposed here, but initially were based on the ebuild by @Vindex17 . I need to test them locally a bit more, but hopefully tomorrow they'll be pushed to the tree with the help of @idella .

**Comment by Coacher** [comment]
Unfortunately I'd rather not commit the ebuild by @Vindex17  because it consists of a single commit with a very huge set of changes without any list what exactly was changed and why. Also verbump + removal in a single commit is not good.

**Comment by idella** [comment]
Vindex17 you have not been ignore it just looks that way. I am second inline to the primary developer and the time has passed now to move on your pull request, however Coacher is now  co-maintainer, he is well versed and what will occur now is the version bump will be based upon your submitted pull request and Coacher will refine and set the final state, and the commit msg he prepares is to ack your contribution. It is now wasted.

**Comment by monsieurp** [comment]
Sorry but this PR is obsolete now :(. Let's focus our efforts on getting PR #544 merged instead.

**Pull request closed without merging**